### ISSUE TITLE
Order: differentiate it's a "sell" from a "buy"

### ISSUE THREAD

**Issue by jfparadis** [Agoric/agoric-sdk#611]
@Chris-Hibbert, @katelynsills, if the JSON structure of an individual order is

[
  {
    "want":{"label":{"assay":{},"allegedName":"moola"},"extent":29},
    "offer":{"label":{"assay":{},"allegedName":"simoleans"},"extent":18}
  }
]

How can we differentiate it's a "sell" from a "buy"?

**Comment by jfparadis** [comment]
BTW, it's kind of a trick question. Might be better to discuss. :)

**Comment by Chris-Hibbert** [comment]
SimpleExchange, like autoSwap doesn't have a natural order. Conventional exchange that people are familiar with usually have currency on one side, and assets (stocks, bonds, mutual funds, cryptocurrency) on the other. In those cases, it makes sense to talk about "buying" and "selling". Here we're mimicking the look and feel, but without the semantics. 

On kraken, I can see UI elements that use green for buy and red for sell, even when both sides are crypto. The UI uses that to say that the denominator should be treated as currency, and prices quoted in units of the denominator.

If we let the user specify the two assets and whether they're buying or selling, we might even be able to show that they're selling ETH for opera ticket seats.

**Comment by katelynsills** [comment]
@jfparadis let me know if I'm understanding you correctly, but I think you're asking not what a sell or buy means in this context, but given an order data structure, how the code could automatically determine whether it is a buy or sell? In that case, can we structure it differently? Maybe:

```
[
   {
    "kind": "sell", 
     "want":{"label":{"assay":{},"allegedName":"moola"},"extent":29},
     "offer":{"label":{"assay":{},"allegedName":"simoleans"},"extent":18}
   }
]
```

(That's probably the wrong way around, but you know what I mean :) )

**Comment by jfparadis** [comment]
I think you are right @katelynsills : on 0x.org there is a named "side" that defines the buy/sell:

```js
        const getColor = (order: OrderBookItem): string => {
            return order.side === OrderSide.Buy ? theme.componentsTheme.green : theme.componentsTheme.red;
        };
```

Is that something we could add?

**Comment by katelynsills** [comment]
That sounds good to me!

**Comment by Chris-Hibbert** [comment]
In the sample data I showed in https://github.com/Agoric/agoric-sdk/issues/541#issuecomment-588577553, one set of orders is labelled 'buys', and the other is labelled 'sells'. Both have 'offers' and 'wants'. They buys all have wants that are in moola, and offers that are in simoleans. The sells are all opposite: the offers are in moola, and wants in simoleans.

Are you saying that the label needs to be inside the orders, or are you saying something else?

**Comment by jfparadis** [comment]
Now, the reason why this is tricky is that it depends on the way the exchange market is set.

On 0x.org, the market is defined by the "currency pair":

```js
{currencyPair.base.toUpperCase()}/{currencyPair.quote.toUpperCase()}
```
See https://github.com/0xProject/0x-launch-kit-frontend/blob/development/src/components/erc20/common/markets_dropdown.tsx#L251

For example, ZRX/ETH is a how the pair is displayed, where ZRX is the "base" and ETH is the "quote".

Still on 0x.org, this distinction carries into the UI in the buy and sell inputs fields:
- the top entry is always the base (ZRX)
- the bottom one is the quote (ETH)
See https://github.com/0xProject/0x-launch-kit-frontend/blob/development/src/components/erc20/marketplace/buy_sell.tsx#L261

For example:

![Screen Shot 2020-02-26 at 5 58 28 PM](https://user-images.githubusercontent.com/2583204/75406276-f0805d00-58c4-11ea-8054-c591770ffbb4.png)


If we ignore the fees, when a user presses "buy" do we get:

```
   {
     "want":{"label":{"assay":{},"allegedName":"ZRX"},"extent":500},
     "offer":{"label":{"assay":{},"allegedName":"ETH"},"extent":590}
   }
```

And  when a user presses "sell":

```
   {
     "offer":{"label":{"assay":{},"allegedName":"ZRX"},"extent":500}
     "want":{"label":{"assay":{},"allegedName":"ETH"},"extent":590},
   }
```

In that case, because the UI is always ZRX/ETH, we don't need the "type" field because we can infer it:
1. "buy" = "want" is the base currency
2. "sell" = "want" is the quote currency

@katelynsills @Chris-Hibbert Makes sense? (that was the trick in the question)

In other words, an exchange must always show the currencies in the same order, it has a sense of the "from" and "to", and that defines what is a "buy" and what is a "sell".

**Comment by Chris-Hibbert** [comment]
Yep. That's the correct interpretation.

**Comment by katelynsills** [comment]
Makes sense to me! I'm fine with a type field as well if that makes things easier, too. The simpleExchange is built under the same concept of "ZRX/ETH" meaning that ZRX is always the underlying asset being traded and ETH is the price, it's just that in our case it's "moola/simoleans". We don't ever switch to simoleans being the underlying asset unless we were to create a second simpleExchange.

**Comment by jfparadis** [comment]
Great, I wished I had understood that from day 1... anyways.

In that case, we do not need this structure `{ buys: [], sells: [] }` because the UI has to flatten it anyways:

```js
  useEffect(() => {
    const result = [];
    orderhistory.buys.forEach(item => result.push({ side: true, ...item }));
    orderhistory.sells.forEach(item => result.push({ side: false, ...item }));
    setHistory(result);
  }, [orderhistory]);
```

**Comment by jfparadis** [comment]
@katelynsills, you can see why we need to get the assays from zoe, and we need them in the right order, so we can define where to place each assay in the buy and sell UI.
